### DISCUSSION TITLE
Use `pyproject.toml` instead of `setuptools`

### DISCUSSION THREAD

**Discussion by karpetrosyan** [encode/httpcore#748] [Ideas]
Should we use pyproject.toml instead of `setuptools`? 
I believe it is preferable to use `pyproject.toml`, as we do for other encode projects.

Refs: https://github.com/encode/httpcore/pull/281

**Comment by lovelydinosaur** [comment]
I'm okay with us mirroring whatever we're doing with `httpx` here.

If you choose to open an issue for this then I'd suggest referencing whichever PR introduced `pyproject.toml` into `httpx` so that we've got some good interlinking context.
